code/mob/living/critter/small_animal.dm

```diff
@@ -1855,6 +1855,150 @@ var/list/mob_bird_species = list("smallowl" = /mob/living/critter/small_animal/b
 		qdel(friends)
 		return ..()
 
+/* =================================================== */
+/* ------------------- Rattlesnake ------------------- */
+/* =================================================== */
+
+/mob/living/critter/small_animal/rattlesnake
+	name = "rattlesnake"
+	real_name = "rattlesnake"
+	blood_id = "blood"
+	desc = "A snake. With a rattle. A rattlesnake."
+	icon_state = "rattlesnake"
+	icon_state_dead = "rattlesnake_dead"
+	speechverb_say = "hisses"
+	speechverb_exclaim = "rattles"
+	speechverb_ask = "hisses"
+	health_brute = 20
+	health_burn = 20
+	density = 0
+	flags = TABLEPASS
+	fits_under_table = 1
+	can_lie = 0
+	ai_type = /datum/aiHolder/rattlesnake
+	is_npc = TRUE
+	var/aggressive = TRUE
+	var/list/friends = list()
+	add_abilities = list(/datum/targetable/critter/wasp_sting/snake_bite)
+
+	New()
+		..()
+		src.event_handler_flags |= USE_PROXIMITY
+
+	setup_hands()
+		..()
+		var/datum/handHolder/HH = hands[1]
+		HH.limb = new /datum/limb/mouth
+		HH.icon = 'icons/mob/critter_ui.dmi'
+		HH.icon_state = "mouth"
+		HH.name = "mouth"
+		HH.limb_name = "teeth"
+		HH.can_hold_items = 0
+
+	attackby(obj/item/I, mob/M)
+		if(istype(I, /obj/item/reagent_containers/food/snacks) && ishuman(M) && !isdead(src))
+			src.visible_message("[M] feeds \the [src] some [I].", "[M] feeds you some [I].")
+			for(var/damage_type in src.healthlist)
+				var/datum/healthHolder/hh = src.healthlist[damage_type]
+				hh.HealDamage(5)
+				src.health_brute = min(60, src.health_brute + 6)
```

Comment:
Indent wrong here? You'd be healing 6 brute/burn damage fro every healthHolder they've got

Reply:
That is the idea, yeah. Like with how on the scorpions you can also get their life up to 60 by giving them food.

Reply:
but what you're doing here is 
```
for type in ("toxic", "brute", "burn", "oxy")
    healdamage(5, type)
    max_health += 6
```
which doesn't seem right at all. It'd seems like what you should do instead (if indeed that's the amount of health you want to give):
```
for type in ("toxic", "brute", "burn", "oxy")
    healdamage(5, type)
max_health += 24
```